run.sh

```diff
@@ -1,3 +1,35 @@
 #!/usr/bin/env bash
 
-go generate ./... && go tool vet -all -shadow=true ./ && go build && ./joker "$@"
+build() {
+    go clean -x
+
+    go generate ./...
+
+    go tool vet -all -shadow=true ./
+
+    go build
+}
+
+set -e  # Exit on error.
+
+build
+
+./joker -e '(print "\nLibraries available in this build:\n  ") *loaded-libs* (println)'
+
+SUM256="$(go run tools/sum256dir/main.go std)"
```

Comment:
> Did you consider using `md5` or `shasum` instead of writing custom tool for this?

Yes, but a) they don't seem to support recursion on directories, and b) I'm trying to avoid adding dependencies on tools that might not be available or consistent (on e.g. Windows or Plan9 or whatever).

Reply:
> > Did you consider using `md5` or `shasum` instead of writing custom tool for this?
> 
> Yes, but a) they don't seem to support recursion on directories, and b) I'm trying to avoid adding dependencies on tools that might not be available or consistent (on e.g. Windows or Plan9 or whatever).

I could be easily convinced to break this out into a separate tool in my own github repo....

Reply:
Another approach might be to just do this in `run.sh`, and leave other-OS compatibility to some other script (like a `run.bat` for Windows):

https://gist.github.com/skratchdot/8053931ee7f3e30dec9d

(That one-liner takes a different approach than mine, but it should be just as effective.)

Reply:
Yep, I was thinking something along those lines. `run.sh` won't run on Windows anyway. As you pointed out, if anyone is ever interested in building Joker on Windows, writing `run.bat` would be the way to go.

Reply:
> Yep, I was thinking something along those lines. `run.sh` won't run on Windows anyway. As you pointed out, if anyone is ever interested in building Joker on Windows, writing `run.bat` would be the way to go.

Actually, I use `run.sh` on Windows, but in the Git Bash environment. I've had to track down and install one or more tools to make that work.

That's why I wanted to keep everything in the "Go world", because the Go ecosystem is pretty good about supporting Windows (and other OSes) in a portable way.

Haven't tested `run.sh` on Windows recently, though...could fire up my Windows machine and try that out on this branch if you'd like.

Reply:
I would not count Git Bash as "Windows". Not sure if `shasum` or `md5` are available there of if they can be installed, but I think it's reasonable for `run.sh` to assume the availability of common tools on POSIX systems.

Reply:
> I would not count Git Bash as "Windows". Not sure if `shasum` or `md5` are available there of if they can be installed, but I think it's reasonable for `run.sh` to assume the availability of common tools on POSIX systems.

In the abstract, I agree.

But, practically speaking, I'm currently _not_ blocked (last I checked) periodically checking to make sure I haven't broken any stuff in Git Bash on Windows, as that's my environment (and it's working).

It might be the one-liner would work just fine on my Git Bash because either Git Bash does indeed have the tools installed or I've since installed them.

But if it doesn't, I'll have to either install the necessary tools, write `run.bat`, or just leave Windows aside for some time until I get around to one of those other things.

I don't think that's a major loss in general, though since my `std` work fundamentally involves handling different targets, it'd be good to ensure I haven't gone too far afield of Windows support.

Reply:
Thank you for the update. Glad `run.sh` works in cygwin.